### PR TITLE
DEVDOCS-5922: [Update] removed 500 status error code from metafield docs

### PR DESCRIPTION
**Pull request by bc-traciporter**
# [DEVDOCS-5922]

## What changed?
Removed 500 status code from metafield docs

Carts
Channels
Customer
Catalog
Orders

## Release notes draft

n/a

* 

## Anything else?

ping {names}

[DEVDOCS-5922]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-5922?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/carts.v3.yml [hunk 1]
```diff
@@ -909,8 +909,6 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorResponse'
-        '500':
```
[the remark this answers is not in the document]
Comment (kskomro) [reply reference/carts.v3.yml:912]: sorry I missed some of the parts in previous PR (it looks like for order batch metafields we still have 500 and for singular delete we are missing 404) - please fix that also

File: reference/orders.v3.yml [hunk 2]
```diff
@@ -969,8 +959,6 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponsePartialSuccess_POST_PUT'
-        '500':
-          description: Internal Server Error
     delete:
       summary: Delete All Metafields
```
Comment (kskomro) [inline reference/orders.v3.yml:963]:
```suggestion
      summary: Delete Multiple Metafields
```

File: reference/carts.v3.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kskomro) [reply reference/carts.v3.yml:912]: @bc-traciporter I noticed that for singular put we also can have 422:
```
‘resourceType/:{resourceId}/metafields/:{metafieldId}’:
   PUT:
	200 - ok
	404 - Not found (A metafield was not found with this query)
        422 - Unprocessable Entity (The following fields are invalid: key)

```

File: reference/carts.v3.yml [hunk 3]
```diff
@@ -895,22 +881,43 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetafieldResponse'
-        '409':
-          description: |
-            The `Metafield` conflicts with another `Metafield`. This can result from duplicate unique key combinations of the appʼs client id, namespace, key, resource_type, and resource_id.
+        '400':
```
Comment (kskomro) [inline reference/carts.v3.yml:884]: I don't think we can get 400 on singular get. Just 200 or 404

Comment (6juara9) [reply reference/carts.v3.yml:884]: Yeah, let's remove all 400's in this PR for get endpoints :)

**Review by kskomro — COMMENTED** [review]
1. Every endpoint that accepts json body can return 400 when json is misformatted.
2. Docs should be consistent for all metafields types.

File: reference/carts.v3.yml [hunk 4]
```diff
@@ -867,19 +853,43 @@ paths:
                   "date_created": 2023-11-15T15:16:35+00:00
                   "date_modified": 2023-11-15T15:16:35+00:00
                 meta: {}
-        '404':
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/carts.v3.yml:857]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/carts.v3.yml [hunk 5]
```diff
@@ -1035,15 +1084,37 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponse_Batch_POST_PUT'
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/carts.v3.yml:1088]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/carts.v3.yml [hunk 6]
```diff
@@ -1076,15 +1147,37 @@ paths:
...
```
Comment (kskomro) [inline reference/carts.v3.yml:1151]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/carts.v3.yml [hunk 7]
```diff
@@ -1076,15 +1147,37 @@ paths:
...
           description: |
             Response object for metafields creation with partial success.
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponsePartialSuccess_POST_PUT'
-        '500':
-          description: Internal Server Error
     delete:
       summary: Delete All Metafields
```
Comment (kskomro) [inline reference/carts.v3.yml:1182]:
```suggestion
      summary: Delete Multiple Metafields
```

File: reference/catalog/brands_catalog.v3.yml [hunk 8]
```diff
@@ -1038,6 +1017,30 @@ paths:
                       date_created: '2021-08-06T19:15:35+00:00'
                       date_modified: '2021-08-06T19:15:35+00:00'
                     meta: {}
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/brands_catalog.v3.yml:1021]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/brands_catalog.v3.yml [hunk 9]
```diff
@@ -1191,6 +1194,30 @@ paths:
                   date_created: '2021-08-06T19:15:35+00:00'
                   date_modified: '2021-08-06T19:15:35+00:00'
                 meta: {}
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/brands_catalog.v3.yml:1198]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/brands_catalog.v3.yml [hunk 10]
```diff
@@ -1399,15 +1445,37 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponse_POST_PUT'
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/brands_catalog.v3.yml:1449]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/brands_catalog.v3.yml [hunk 11]
```diff
@@ -1442,15 +1510,37 @@ paths:
...
```
Comment (kskomro) [inline reference/catalog/brands_catalog.v3.yml:1514]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/brands_catalog.v3.yml [hunk 12]
```diff
@@ -1442,15 +1510,37 @@ paths:
...
           description: |
             Response object for metafields creation with partial success.
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponsePartialSuccess_POST_PUT'
-        '500':
-          description: Internal Server Error
     delete:
       summary: Delete All Metafields
```
Comment (kskomro) [inline reference/catalog/brands_catalog.v3.yml:1545]:
```suggestion
      summary: Delete Multiple Metafields
```

File: reference/catalog/categories_catalog.v3.yml [hunk 13]
```diff
@@ -1052,6 +1031,30 @@ paths:
                   date_created: '2021-08-06T19:15:35+00:00'
                   date_modified: '2021-08-06T19:15:35+00:00'
                 meta: {}
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/categories_catalog.v3.yml:1035]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/categories_catalog.v3.yml [hunk 14]
```diff
@@ -1204,6 +1207,30 @@ paths:
...
```
Comment (kskomro) [inline reference/catalog/categories_catalog.v3.yml:1211]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/categories_catalog.v3.yml [hunk 15]
```diff
@@ -1495,15 +1541,37 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponse_POST_PUT'
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/categories_catalog.v3.yml:1545]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/categories_catalog.v3.yml [hunk 16]
```diff
@@ -1538,15 +1606,37 @@ paths:
...
```
Comment (kskomro) [inline reference/catalog/categories_catalog.v3.yml:1610]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/categories_catalog.v3.yml [hunk 17]
```diff
@@ -1538,15 +1606,37 @@ paths:
...
           description: |
             Response object for metafields creation with partial success.
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponsePartialSuccess_POST_PUT'
-        '500':
-          description: Internal Server Error
     delete:
       summary: Delete All Metafields
```
Comment (kskomro) [inline reference/catalog/categories_catalog.v3.yml:1641]:
```suggestion
      summary: Delete Multiple Metafields
```

File: reference/catalog/product-variants_catalog.v3.yml [hunk 18]
```diff
@@ -630,6 +609,30 @@ paths:
                   date_created: '2021-08-06T19:15:35+00:00'
                   date_modified: '2021-08-06T19:15:35+00:00'
                 meta: {}
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/product-variants_catalog.v3.yml:613]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/product-variants_catalog.v3.yml [hunk 19]
```diff
@@ -784,6 +787,30 @@ paths:
                   date_created: '2018-09-13T16:42:37+00:00'
                   date_modified: '2018-09-13T16:42:37+00:00'
                 meta: {}
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/product-variants_catalog.v3.yml:791]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/product-variants_catalog.v3.yml [hunk 20]
```diff
@@ -1664,15 +1710,37 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponse_POST_PUT'
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/product-variants_catalog.v3.yml:1714]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/product-variants_catalog.v3.yml [hunk 21]
```diff
@@ -1707,16 +1775,37 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponse_POST_PUT'
-
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/product-variants_catalog.v3.yml:1779]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/products_catalog.v3.yml [hunk 22]
```diff
@@ -4149,6 +4128,30 @@ paths:
                   date_created: '2018-09-13T16:42:37+00:00'
                   date_modified: '2018-09-13T16:42:37+00:00'
                 meta: {}
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/products_catalog.v3.yml:4132]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/products_catalog.v3.yml [hunk 23]
```diff
@@ -4301,6 +4304,30 @@ paths:
                   date_created: '2021-08-06T19:15:35+00:00'
                   date_modified: '2021-08-06T19:15:35+00:00'
                 meta: {}
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/products_catalog.v3.yml:4308]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/products_catalog.v3.yml [hunk 24]
```diff
@@ -5222,15 +5268,37 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponse_POST_PUT'
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/catalog/products_catalog.v3.yml:5272]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/products_catalog.v3.yml [hunk 25]
```diff
@@ -5263,15 +5331,37 @@ paths:
...
```
Comment (kskomro) [inline reference/catalog/products_catalog.v3.yml:5335]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/catalog/products_catalog.v3.yml [hunk 26]
```diff
@@ -5263,15 +5331,37 @@ paths:
...
           description: |
             Response object for metafields creation with partial success.
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponsePartialSuccess_POST_PUT'
-        '500':
-          description: Internal Server Error
     delete:
       summary: Delete All Metafields
```
Comment (kskomro) [inline reference/catalog/products_catalog.v3.yml:5366]:
```suggestion
      summary: Delete Multiple Metafields
```

File: reference/channels.v3.yml [hunk 27]
```diff
@@ -702,6 +702,44 @@ paths:
       responses:
         '200':
           $ref: '#/components/responses/metafield_200'
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/channels.v3.yml:706]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/channels.v3.yml [hunk 28]
```diff
@@ -740,6 +791,43 @@ paths:
...
```
Comment (kskomro) [inline reference/channels.v3.yml:795]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/channels.v3.yml [hunk 29]
```diff
@@ -809,15 +908,36 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponse_POST_PUT'
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/channels.v3.yml:912]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/channels.v3.yml [hunk 30]
```diff
@@ -840,15 +960,37 @@ paths:
...
```
Comment (kskomro) [inline reference/channels.v3.yml:964]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/channels.v3.yml [hunk 31]
```diff
@@ -840,15 +960,37 @@ paths:
...
           description: |
             Response object for metafields creation with partial success.
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponsePartialSuccess_POST_PUT'
-        '500':
-          description: Internal Server Error
     delete:
       summary: Delete All Metafields
```
Comment (kskomro) [inline reference/channels.v3.yml:995]:
```suggestion
      summary: Delete Multiple Metafields
```

File: reference/customers.v3.yml [hunk 32]
```diff
@@ -1704,6 +1702,43 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionPostPutResponses'
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/customers.v3.yml:1706]:
```suggestion
          description: Bad Request. Inupt is invalid.
```

File: reference/customers.v3.yml [hunk 33]
```diff
@@ -1773,6 +1821,43 @@ paths:
...
```
Comment (kskomro) [inline reference/customers.v3.yml:1825]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/customers.v3.yml [hunk 34]
```diff
@@ -1852,15 +1948,37 @@ paths:
...
```
Comment (kskomro) [inline reference/customers.v3.yml:1952]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/customers.v3.yml [hunk 35]
```diff
@@ -1892,15 +2010,37 @@ paths:
...
```
Comment (kskomro) [inline reference/customers.v3.yml:2014]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/customers.v3.yml [hunk 36]
```diff
@@ -1892,15 +2010,37 @@ paths:
...
           description: |
             Response object for metafields creation with partial success.
           content:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponsePartialSuccess_POST_PUT'
-        '500':
-          description: Internal Server Error
     delete:
       summary: Delete All Metafields
```
Comment (kskomro) [inline reference/customers.v3.yml:2045]:
```suggestion
      summary: Delete Multiple Metafields
```

File: reference/orders.v3.yml [hunk 37]
```diff
@@ -620,6 +607,30 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetafieldResponse'
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/orders.v3.yml:611]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/orders.v3.yml [hunk 38]
```diff
@@ -694,6 +705,30 @@ paths:
...
```
Comment (kskomro) [inline reference/orders.v3.yml:709]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/orders.v3.yml [hunk 39]
```diff
@@ -921,15 +961,37 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/MetaFieldCollectionResponse_POST_PUT'
+        '400':
+          description: Bad Request. Metafield is invalid.
```
Comment (kskomro) [inline reference/orders.v3.yml:965]:
```suggestion
          description: Bad Request. Input is invalid.
```

File: reference/orders.v3.yml [hunk 40]
```diff
@@ -962,17 +1024,39 @@ paths:
...
```
Comment (kskomro) [inline reference/orders.v3.yml:1028]:
```suggestion
          description: Bad Request. Input is invalid.
```

**Review by kskomro — APPROVED** [review]

**Review by kskomro — APPROVED** [review]

**Review by bc-andreadao — APPROVED** [review]

**Pull request merged**